pallets/mnt-token/src/lib.rs

```diff
@@ -294,37 +295,43 @@ impl<T: Config> Pallet<T> {
 // RPC methods
 impl<T: Config> Pallet<T> {
 	/// Gets MNT accrued but not yet transferred to user.
-	pub fn get_unclaimed_mnt_balance(account_id: &T::AccountId) -> BalanceResult {
-		let accrued_mnt =
-			MntSpeeds::<T>::iter().try_fold(Balance::zero(), |current_accrued, (pool_id, _)| -> BalanceResult {
-				Self::update_mnt_borrow_index(pool_id)?;
+	///
+	/// - `user`: user id.
+	pub fn get_user_total_unclaimed_mnt_balance(account_id: &T::AccountId) -> BalanceResult {
+		let accrued_mnt = MntSpeedStorage::<T>::iter().try_fold(
+			Balance::zero(),
+			|current_accrued, (pool_id, _)| -> BalanceResult {
+				Self::pool_update_mnt_borrow_index(pool_id)?;
 				let accrued_borrow_mnt = Self::distribute_borrower_mnt(pool_id, account_id, true)?;
-				Self::update_mnt_supply_index(pool_id)?;
+				Self::pool_update_mnt_supply_index(pool_id)?;
 				let accrued_supply_mnt = Self::distribute_supplier_mnt(pool_id, account_id, true)?;
 				Ok(current_accrued + accrued_borrow_mnt + accrued_supply_mnt)
-			})?;
+			},
+		)?;
 		Ok(accrued_mnt)
 	}
 }
 
 impl<T: Config> MntManager<T::AccountId> for Pallet<T> {
 	/// Update mnt supply index for pool
 	/// Do nothing in case if distribution is off or index have already been updated in this block.
-	fn update_mnt_supply_index(underlying_id: CurrencyId) -> DispatchResult {
+	///
+	/// - `pool_id`: id of the pool to update index
+	fn pool_update_mnt_supply_index(pool_id: CurrencyId) -> DispatchResult {
```

Comment:
```suggestion
	fn update_pool_mnt_supply_index(pool_id: CurrencyId) -> DispatchResult {
```
Maybe this is better?